packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -90,4 +90,17 @@ class UrlLauncherPlugin extends UrlLauncherPlatform {
     openNewWindow(url, webOnlyWindowName: webOnlyWindowName);
     return true;
   }
+
+  @override
+  Future<bool> supportsMode(PreferredLaunchMode mode) async {
+    // Web doesn't allow any control over the destination beyond
+    // webOnlyWindowName, so don't claim support for any mode beyond default.
+    return mode == PreferredLaunchMode.platformDefault;
```

Comment:
These are good defaults for the web platform. This maybe could be expanded more (if you use window.open, and keep a reference to the window that you opened, it can be [`close`](https://developer.mozilla.org/en-US/docs/Web/API/Window/close)d, for example), but so far I have never heard from anybody wanting the "close" feature on the web.